### ISSUE TITLE
Enhance form functions for accessibility

### ISSUE THREAD

**Issue by lizardat** [contao/contao#6976]
### Description

Contao already provides a very good base for accessible websites. To further enhance the possibilities, I would suggest these additions to the form module:

**autocomplete**

One criteria for accessible websites is the usage of "autocomplete" with input fields in forms. 

It would be great help if the backend offers a select field for autocomplete for `<input>`, `<select>` and `<textarea>`. Possible options are listed at [https://www.w3.org/TR/WCAG21/#input-purposes](https://www.w3.org/TR/WCAG21/#input-purposes)

`<form>` should get the choice of options autocomplete=on/off

**novalidate**

Furthermore, `<form>` should be able to get "novalidate" to prevent validation of the form input by browsers but leave the validation fully to the settings from Contao.

**aria in case of error**

input fields should get a aria-describedby connected to the error message in case of an validation error.

Error messages should be marked with role="alert" and aria-live="assertive"

**Comment by fritzmg** [comment]
> __novalidate__
>
> Furthermore, `<form>` should be able to get "novalidate" to prevent validation of the form input by browsers but leave the validation fully to the settings from Contao.

This you can already configure in the form's settings. Though why would you enable that for the purpose of accessibility?

**Comment by lizardat** [comment]
Browser validation is sufficient for WCAG 2.1 3.3.1 Error Identification

> If an input error is automatically detected, the item that is in error is identified and the error is described to the user in text.

but does not provide enough information as requested in 3.3.3 Error Suggestion

> If an input error is automatically detected and suggestions for correction are known, then the suggestions are provided to the user, unless it would jeopardize the security or purpose of the content.

**Comment by asaage** [comment]
> Browser validation is sufficient

Funfact: chromes validation accepts ```mail@example``` (without a tld) in <input type="email">

**Comment by fritzmg** [comment]
> but does not provide enough information as requested in 3.3.3 Error Suggestion

In what way does the browser validation not provide enough information?

**Comment by aschempp** [comment]
> Funfact: chromes validation accepts ```mail@example``` (without a tld) in ```<input type="email">```

Because that is a valid email address... e.g. `you@localhost`

**Comment by asaage** [comment]
> Because that is a valid email address

Yep - but contao wont't like it.

**Comment by fritzmg** [comment]
@asaage pls discuss this in a separat issue, not here.

**Comment by lizardat** [comment]
> In what way does the browser validation not provide enough information?

Because "Feld ausfüllen" ist not sufficient help to correct errors, 
see [https://www.w3.org/WAI/WCAG20/Understanding/error-suggestion](https://www.w3.org/WAI/WCAG20/Understanding/error-suggestion)

**Comment by fritzmg** [comment]
> Because "Feld ausfüllen" ist not sufficient help to correct errors,
> see https://www.w3.org/WAI/WCAG20/Understanding/error-suggestion

If you think this is not enough (I don't really see why that would be the case from your provided link) then this is something you should report to the respective Browser vendors. Generally I do not see how _disabling_ browser validation increases accessibility as the browser can help the user better and more immediately (with enough information provided). In any case as per your original request and as already mentioned - this is something you can already do.

**Comment by leofeyer** [comment]
As discussed in the Contao call,

* we want to support [autocomplete attributes](https://www.w3.org/TR/WCAG21/#input-purposes) in the form generator;
* `novalidate` can be enabled in the form settings already;
* we want to research to what extent ARIA attributes should be set for error messages.

**Comment by lizardat** [comment]
> * we want to research to what extent ARIA attributes should be set for error messages.

I would like to contribute this reference which was built in accordance with one of Austrias main web accessibility experts: 
https://www.digitalbarrierefrei.at/de/beschwerdestelle/kontaktformular-beschwerdestelle
always open for further improvement...

**Comment by zoglo** [comment]
> we want to support [autocomplete attributes](https://www.w3.org/TR/WCAG21/#input-purposes) in the form generator

Done in #7473

>     * `novalidate` can be enabled in the form settings already;

Has already been possible

>     * we want to research to what extent ARIA attributes should be set for error messages.

https://github.com/contao/contao/issues/6976#issuecomment-1983240397

This should be possible with additional help texts -> #7825 

@lizardat feel free to open new issues with detailed examples on how to implement new form features + their WCAG Guidelines.

**Comment by zoglo** [comment]
> I don't think we should close this, because the following is definitely not the case currently:

I opened two sub issues to track the 2 remaining feature requests:

- [#9176](https://github.com/contao/contao/issues/9176)
- [#9175](https://github.com/contao/contao/issues/9175)
